internal/credential/credential_libraries.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// CredentialLibraryRepository defines the interface expected
+// to get the total number of credential libraries and deleted ids.
+type CredentialLibraryRepository interface {
```

Comment:
```suggestion
type LibraryRepository interface {
```
Wouldn't `credential.LibraryRepository` be more like the existing interfaces in this package?

Reply:
Sure, that sounds good